packages/orval/src/client.ts

```diff
@@ -35,6 +36,13 @@ export const GENERATOR_CLIENT: GeneratorClients = {
   zod: zod()(),
 };
 
+const ORVAL_DEPENDENCIES: GeneratorDependency[] = [
```

Comment:
I don't know about importing orval as a dependency of the generated code. You create then a direct dependence on orval. Would make more sense to generate it automatically with all the other schemas no?

Reply:
Yes, I think that would be better specifically if Orval is only a dev dependency. The only question that I then have how should we distinguish this "orval defined" schema from "specification defined" schemas? Because I don't know whether it would be acceptable to mix these types among all other schemas?

What do you think would be a proper place to put these types @anymaniax?

Reply:
Good question because in that case, we could have some conflicts with the generated models from the specification. Do you know that you can already achieve that with a mutator and exporting a BodyType in it? 

``` ts
export type BodyType<Body> = NonReadonly<Body>;
```

Reply:
I did not know this then technically speaking the issue can be solved by simply setting the mutator correctly.
```typescript
// https://stackoverflow.com/questions/49579094/typescript-conditional-types-filter-out-readonly-properties-pick-only-requir/49579497#49579497
type IfEquals<X, Y, A = X, B = never> = (<T>() => T extends X ? 1 : 2) extends <
  T,
>() => T extends Y ? 1 : 2
  ? A
  : B;

type WritableKeys<T> = {
  [P in keyof T]-?: IfEquals<
    { [Q in P]: T[P] },
    { -readonly [Q in P]: T[P] },
    P
  >;
}[keyof T];

type Writable<T> = Pick<T, WritableKeys<T>>;
export type NonReadonly<T> = {
  [P in keyof Writable<T>]: T[P] extends number | string | boolean
    ? T[P]
    : NonReadonly<NonNullable<T[P]>>;
};

export type BodyType<Body> = NonReadonly<Body>;
```

Reply:
Is this small use-case then important enough to change Orval to adhere to this OpenAPI specification? Or should this be the solution to issue #616 ?